libbeat/processors/convert/convert.go

```diff
@@ -0,0 +1,366 @@
+	case uint:
+		return int32(v), nil
+	case uint8:
+		return int32(v), nil
+	case uint16:
+		return int32(v), nil
+	case uint32:
+		return int32(v), nil
+	case uint64:
+		return int32(v), nil
+	case float32:
+		return int32(v), nil
+	case float64:
+		return int32(v), nil
+	default:
+		return 0, errors.Errorf("invalid conversion of [%T] to integer", value)
+	}
+}
+
+func toFloat(value interface{}) (float32, error) {
+	switch v := value.(type) {
+	case string:
+		f, err := strconv.ParseFloat(v, 32)
+		return float32(f), err
+	case int:
+		return float32(v), nil
+	case int8:
+		return float32(v), nil
+	case int16:
+		return float32(v), nil
+	case int32:
+		return float32(v), nil
+	case int64:
+		return float32(v), nil
+	case uint:
+		return float32(v), nil
+	case uint8:
+		return float32(v), nil
+	case uint16:
+		return float32(v), nil
+	case uint32:
+		return float32(v), nil
+	case uint64:
+		return float32(v), nil
+	case float32:
+		return v, nil
+	case float64:
+		return float32(v), nil
+	default:
+		return 0, errors.Errorf("invalid conversion of [%T] to float", value)
+	}
+}
+
+func toDouble(value interface{}) (float64, error) {
+	switch v := value.(type) {
+	case string:
+		f, err := strconv.ParseFloat(v, 64)
+		return float64(f), err
+	case int:
+		return float64(v), nil
+	case int8:
+		return float64(v), nil
+	case int16:
+		return float64(v), nil
+	case int32:
+		return float64(v), nil
+	case int64:
+		return float64(v), nil
+	case uint:
+		return float64(v), nil
+	case uint8:
+		return float64(v), nil
+	case uint16:
+		return float64(v), nil
+	case uint32:
+		return float64(v), nil
+	case uint64:
+		return float64(v), nil
+	case float32:
+		return float64(v), nil
+	case float64:
+		return v, nil
+	default:
+		return 0, errors.Errorf("invalid conversion of [%T] to float", value)
+	}
+}
+
+func toBoolean(value interface{}) (bool, error) {
+	switch v := value.(type) {
+	case string:
+		return strconv.ParseBool(v)
+	case bool:
+		return v, nil
+	default:
+		return false, errors.Errorf("invalid conversion of [%T] to boolean", value)
+	}
+}
+
+func toIP(value interface{}) (string, error) {
+	switch v := value.(type) {
+	case string:
+		// This is validating that the value is an IP.
+		if net.ParseIP(v) != nil {
+			return v, nil
+		}
+	}
+	return "", errors.Errorf("invalid conversion of [%T] to IP", value)
+}
+
+func annotateError(id string, err error) error {
+	if err == nil {
+		return nil
+	}
+	if id != "" {
+		return errors.Wrapf(err, "failed in processor.convert with instance_id=%v", id)
+	}
+	return errors.Wrap(err, "failed in processor.convert")
+}
+
+// cloneValue returns a shallow copy of a map. All other types are passed
+// through in the return. This should be used when making straight copies of
+// maps without doing any type conversions.
+func cloneValue(value interface{}) interface{} {
+	switch v := value.(type) {
+	case common.MapStr:
+		return v.Clone()
+	case map[string]interface{}:
+		return common.MapStr(v).Clone()
+	default:
```

Comment:
Here, @urso [suggests](https://github.com/elastic/beats/pull/11761#discussion_r280467032) adding a case for arrays too.

I [added](https://github.com/elastic/beats/pull/11761/commits/03adad0aaad4c54145dca2bf754ddc21cf0401cc) a simple case for []interface{}

Reply:
Thanks, I copied that in.